lms/djangoapps/mobile_api/video_outlines/views.py

```diff
@@ -118,7 +119,25 @@ def get(self, request, course, *args, **kwargs):
             video_descriptor = modulestore().get_item(usage_key)
             transcripts = video_descriptor.get_transcripts_info()
             content, filename, mimetype = video_descriptor.get_transcript(transcripts, lang=lang)
-        except (NotFoundError, ValueError, KeyError):
+        except NotFoundError:
+            # Fallback mechanism for edx-val transcripts
+            transcript = get_video_transcript_content(
+                course_id=usage_key.course_key,
+                language_code=lang,
+                edx_video_id=video_descriptor.edx_video_id,
+                youtube_id_1_0=video_descriptor.youtube_id_1_0,
+                html5_sources=video_descriptor.html5_sources,
+            )
+            if not transcript:
+                raise Http404(u"Transcript not found for {}, lang: {}".format(block_id, lang))
+
+            content = Transcript.convert(transcript['content'], 'sjson', 'srt')
+            filename = '{filename}.{ext}'.format(
+                filename=transcript['file_name'].split('.')[0],
```

Comment:
This will not work if filename contains `.` characters.
```bash
>>> 'a.b.c.srt'.split('.')[0]
'a'
```
We can use os module to get filename
```bash
>>> import os
>>> os.path.splitext(os.path.basename('a.b.c.srt'))[0]
'a.b.c'
```